services/importService.js

```diff
@@ -0,0 +1,72 @@
+const axios = require('axios')
+const Product = require('../models/Product')
+
+const path = require('path')
+require('dotenv').config({ path: path.resolve(__dirname, '../.env') })
+const { SNEAKER_DB_URL } = process.env
+
+const shopUrl = SNEAKER_DB_URL
+const pageLimit = 313
+let pageCounter = 0
+
+const getProducts = async url => {
+  axios
+    .get(url)
+    .then(res => {
+      const products = res.data.results
+      products
+        .filter(product => product.retailPrice !== '' || null || undefined)
+        .filter(product => product.styleId !== '' || null || undefined)
+        .filter(product => product.media.imageUrl !== null || undefined)
+        .filter(product => product.media.smallImageUrl !== null || undefined)
+        .filter(product => product.media.thumbUrl !== null || undefined)
```

Comment:
Maybe you could refactor this. A lot of repetition in there.

Reply:
any idea? thought about it, but found no way to solve it 🤷‍♂️

Reply:
Could you show a sample of the products object which you get from your fetch?

Reply:
{
      "id": "02600cb6-9369-46ca-93fb-a1ade2212ea4",
      "brand": "Jordan",
      "colorway": "Bordeaux/White-Metallic Silver",
      "gender": "men",
      "name": "Bordeaux",
      "releaseDate": "2021-10-20",
      "retailPrice": 170,
      "shoe": "Jordan 1 Retro High",
      "styleId": "555088-611",
      "title": "Jordan 1 Retro High Bordeaux",
      "year": 2021,
      "media": {
        "imageUrl": null,
        "smallImageUrl": null,
        "thumbUrl": null
      }
    },

Reply:
> any idea? thought about it, but found no way to solve it 🤷‍♂️

Hey @marcushohlbein sorry for the late reply.
Today I found an interesting article which seems to do exactly what we were looking for 😉

https://www.freecodecamp.org/news/supercharged-sorts-in-javascript/